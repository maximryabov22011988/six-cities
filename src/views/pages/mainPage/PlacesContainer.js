import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  leftPanel: PropTypes.element,
  rightPanel: PropTypes.element
};

const defaultProps = {
  leftPanel: null,
  rightPanel: null
};

function PlacesContainer({ leftPanel, rightPanel }) {
  return (
    <div className="cities__places-wrapper">
      <div className="cities__places-container container">
        {leftPanel}
        <div className="cities__right-section">
          <section className="cities__map map">{rightPanel}</section>
        </div>
      </div>
    </div>
  );
}

PlacesContainer.propTypes = propTypes;
PlacesContainer.defaultProps = defaultProps;

export default PlacesContainer;
