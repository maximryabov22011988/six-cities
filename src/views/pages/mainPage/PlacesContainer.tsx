import * as React from 'react';
import cn from 'classnames';

interface Props {
  isEmpty: boolean;
  leftPanel: React.ReactNode;
  rightPanel: React.ReactNode;
}

function PlacesContainer({ isEmpty, leftPanel = null, rightPanel = null }: Props) {
  return (
    <div className="cities__places-wrapper">
      <div className={cn('cities__places-container container', isEmpty && 'cities__places-container--empty')}>
        {leftPanel}
        <div className="cities__right-section">
          {!isEmpty && <section className="cities__map map">{rightPanel}</section>}
        </div>
      </div>
    </div>
  );
}

export default PlacesContainer;
