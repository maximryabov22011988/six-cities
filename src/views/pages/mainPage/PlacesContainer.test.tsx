import * as React from 'react';
import * as renderer from 'react-test-renderer';

import PlacesContainer from './PlacesContainer';

it('PlacesContainer renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <PlacesContainer
        isEmpty={false}
        leftPanel={null}
        rightPanel={null}
      />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
