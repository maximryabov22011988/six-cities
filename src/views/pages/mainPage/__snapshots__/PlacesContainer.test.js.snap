// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesContainer correctly render 1`] = `
<div
  className="cities__places-wrapper"
>
  <div
    className="cities__places-container container"
  >
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
      />
    </div>
  </div>
</div>
`;
